Shrink DB Size - Convert event_type to int

mjiori opened the pull request:
In order to shrink the Postgres database, changes need to be made to code referencing the PREMIS event's event_type. The event_type will now be stored as an int, not a string, to save space. We can also save these values as constants, and continue to use them in code.

mjiori committed:
Changing premis event event type from string to int

mjiori committed:
Fixing regressions

diamondap reviewed (commented):
Looks good so far. See comment on util/util.go. Looks like this is a work in progress and we can't fully test it until other PRs related to db changes have been merged. We'll circle back to this one later.

Caesonia committed:
Updating the CFN default template for the make update-template command so we do not wipe out the CFN changes and cause regression.

Caesonia committed:
Merge pull request #194 from APTrust/pause-uploader-scale

Updating the CFN default template for the make update-template comman…

diamondap committed:
Upgraded to minio-go client version 7.0.98 with support for 50TB S3 files.

This update removes our custom DefaultMinioPutOptions in favor of the standard minio.PutObjectOptions{}. Our former DefaultMinioPutOptions were experimental and included settings to work with Wasabi. We are no longer using Wasabi, so we no longer need those settings. In addition, those old Wasabi-specific settings cause problems with the latest minio-go client when uploading files of unknown size (size = -1) because the old Wasabi-specific settings, with their hard-coded checksum algorithm, prevented minio-go from choosing the proper checksum algorithm for files of unknown size. This resulted in failed uploads. https://trello.com/c/53T5k08v

diamondap committed:
When constructing RegistryClient, allow caller to specify endpoint prefix.

The endpoint prefix for preservation services will always be "admin-api." This fix is for DART 3, which borrows the RegistryClient and needs to call "member-api." https://trello.com/c/xU77Ploy

diamondap committed:
Merge branch 'master' into s3-large-files

diamondap committed:
Added a comment re: possibly unused constant

diamondap committed:
Merge branch 'master' into s3-large-files

Caesonia committed:
Merge pull request #196 from APTrust/s3-large-files

S3 large files merges the latest minio client for s3 objects. The primary change for APTrut on this merge is increasing the S3 object size allowed from 5TB to 50TBs.

Caesonia committed:
Just commiting the cfn template for this branch.

Caesonia committed:
updating the template for the latest version.

diamondap committed:
Turn on S3 tracing for 7TB Miami bag

diamondap committed:
Explicitly name os.Stdout as output stream for Minio tracing

Caesonia committed:
Updating template changes.

Caesonia committed:
Updating template.

diamondap committed:
Reverted Minio tracing for Miami bag

diamondap committed:
MetadataGatherer now uses new GetLargeObject call to download bags larger than 5TB

Caesonia committed:
Merge pull request #197 from APTrust/large-bag-reader

MetadataGatherer now uses new GetLargeObject call to download bags la…

Caesonia committed:
Updating cfn templates.

diamondap committed:
StagingUploader now also uses GetLargeObject for files over 5TB

Caesonia committed:
Merge pull request #198 from APTrust/large-bag-reader

StagingUploader now also uses GetLargeObject for files over 5TB

Caesonia committed:
Updating template for minio-deployment.

diamondap committed:
Converting Ruby scripts to bash. e2e tests are not yet passing.

diamondap committed:
e2e tests now pass with test.sh

diamondap committed:
test.sh now properly kills registry when exiting

diamondap committed:
Replaced build.rb and test.rb with build.sh and test.sh in docs

diamondap committed:
Removed old Ruby scripts that have been replaced with bash scripts.

mjiori committed:
Fixed regressions in unit tests now working on integration

Caesonia committed:
Downscaling the uploader store based on recommendations and because we stop automated scaling.

Caesonia committed:
Deployment for container vulnerability update.

mjiori committed:
Merge branch 'master' into shrink-db-size

mjiori committed:
Merge branch 'master' into shrink-db-size

mjiori committed:
Fixing E2E tests that broke with the new changes

mjiori commented on util/util.go:
Done.

diamondap reviewed (approved):
This all looks good.
